lib/cloud/aws/identity.go

```diff
@@ -52,6 +61,20 @@ type Role struct {
 	identityBase
 }
 
+// String returns the AWS identity ARN for the role. If the role is
+// an assumed-role, it is converted to the IAM role ARN instead of the
+// assumed-role ARN. This is necessary since identity.String() is used in
+// AWS IAM policy statements as the resource, and assumed-role is invalid
+// for those statements.
+func (r Role) String() string {
+	if r.GetType() == resourceTypeRole {
+		return r.identityBase.String()
+	}
+	arn := r.identityBase.arn
+	arn.Resource = fmt.Sprintf("%s/%s", resourceTypeRole, r.GetName())
```

Comment:
I believe in rare cases where the role has a path like `arn:aws:iam::xxxxxx:role/aa/bb/cc`, the assumed-role will be `arn:aws:sts::xxxxxx:assumed-role/cc/<session-name>`, the original role cannot be re-constructed. But at least in our dev account those roles with paths are service roles or reserved roles so I don't think an agent can use one.

Is `String()` used for printing or we actually trying to convert an assumed role to role?

Reply:
went with the above in 3eafd129a94d83092762ca82fd7add29a1ee4b6e

and reverted semaphore name in b51bf8bbdeba0c2f1e64f1f3d245f3ec679b3381